spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/spring/MetricCollectingConfigurator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.spring;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.AnnotationAwareOrderComparator;
+import org.springframework.core.annotation.Order;
+
+import com.linecorp.armeria.common.metric.AbstractMetricCollectingBuilder;
+import com.linecorp.armeria.server.metric.MetricCollectingService;
+
+/**
+ * Interface used to configure a {@code MetricCollectingService} on the default Armeria server.
+ */
+@FunctionalInterface
+public interface MetricCollectingConfigurator extends Ordered {
+
+    /**
+     * Configures the {@link MetricCollectingService} using the specified {@link AbstractMetricCollectingBuilder}.
+     */
+    void configure(AbstractMetricCollectingBuilder metricCollectingServiceBuilder);
```

Comment:
Although we only configure `MetricCollectingServiceBuilder` in `ArmeriaConfigurationUtil.java` I'm using the super class here. `MetricCollectingClientBuilder` and `MetricCollectingRpcClientBuilder` can be configured separately

Reply:
> MetricCollectingClientBuilder and MetricCollectingRpcClientBuilder can be configured separately

Do you want to configure the client builders with `@Qualifier` annotation? I prefer to use `MetricCollectingServiceBuilder` for clarification. I think this should be only used for `MetricCollectingService`.

Reply:
Hmm yes you are right, I want to let users use the builder classes and not the `@Qualifier` annotation. So I changed the type here 👍